### PR TITLE
New function to read version etag 

### PR DESCRIPTION
**Pull request by rafahop**
### What

- Add new function returning headers for Get version. Currently only containing etag
- Add unit test for the happy path of GetVersion and for the new function GetVersionWithHeaders
- Refactor checks of the response headers in unit test so to expand it to collection id, florence token and download service token.

### How to review

Check the code and test look ok

See https://github.com/ONSdigital/dp-dataset-api/pull/416 for the change in dataset api introducing the etag in the header

### Who can review

Anyone.

### CODE REVIEW

File: dataset/dataset_test.go [hunk 1]
```diff
@@ -221,11 +236,98 @@ func TestClient_HealthChecker(t *testing.T) {
...
+					ServiceToken:         serviceAuthToken,
+					CollectionId:         collectionID,
+					DownloadServiceToken: downloadServiceAuthToken,
+				}
+				checkRequestBase(httpClient, http.MethodGet, expectedUrl, expectedHeaders)
+			})
+		})
+
+		Convey("when GetVersionWithHeaders is called", func() {
+			got, h, err := datasetClient.GetVersionWithHeaders(ctx, userAuthToken, serviceAuthToken, downloadServiceAuthToken, collectionID, datasetId, edition, versionString)
```
Comment (s-xton) [inline dataset/dataset_test.go:296]: It is your intension that the THEN clauses are tested in isolation with multiple calls to GetVersionWithHeaders ?

**Review by s-xton — APPROVED** [review]

**Review by s-xton — CHANGES_REQUESTED** [review]
see code comment

File: dataset/dataset_test.go (hunk 1 above)
Comment (rafahop) [reply dataset/dataset_test.go:296]: Not really, but thought it'd read better using separate conveys but I'll join them in one to improve performance 👍

**Review by s-xton — APPROVED** [review]
can you raise a story to move the if statement on the service token to another function pls, not needed now but nice to clean up later

**Pull request merged**